integration-tests/suites/duplicate_endpoints.go

```diff
@@ -93,27 +65,17 @@ func (s *DuplicateEndpointsTestSuite) SetupSuite() {
 	_, err = s.execContainer("socat", command)
 
 	time.Sleep(20 * time.Second)
-
-	err = s.collector.TearDown()
-	s.Require().NoError(err)
-
-	s.db, err = s.collector.BoltDB()
-	s.Require().NoError(err)
 }
 
 func (s *DuplicateEndpointsTestSuite) TearDownSuite() {
+	s.StopCollector()
 	s.cleanupContainer([]string{"socat", "collector"})
 	stats := s.GetContainerStats()
 	s.PrintContainerStats(stats)
 	s.WritePerfResults("DuplicateEndpoints", stats, s.metrics)
 }
 
 func (s *DuplicateEndpointsTestSuite) TestDuplicateEndpoints() {
-	processes, err := s.GetProcesses(s.serverContainer)
-	s.Require().NoError(err)
-	endpoints, err := s.GetEndpoints(s.serverContainer)
-	s.Require().NoError(err)
-
-	assert.Equal(s.T(), 2, len(endpoints))
-	assert.Equal(s.T(), 10, len(processes))
+	s.Sensor().ExpectEndpointsN(s.T(), s.serverContainer, 10*time.Second, 2)
+	s.Sensor().ExpectProcessesN(s.T(), s.serverContainer, 10*time.Second, 8)
```

Comment:
The sleep here should be 20 seconds, like it was in the original test. We need to make sure that at the end of the scrape interval there are no extra endpoints reported. Also we should simply assert that after 20 seconds we have the endpoints and processes that we expect. We should not end the test early if we get the endpoints that we expect. There might be additional unexpected endpoints reported. The point of this test was that we were getting duplicate endpoints reported and this test was supposed to catch that bug. I think the way you have this now might not catch the bug this test was intended to catch.

Reply:
To be clear, this is not a sleep, but a worst-case limit to wait for additional endpoints/processes. The original 20 second sleep remains in the SetupSuite of this test. 

I'll add some additional asserts here to verify that no more processes or endpoints are reported than the ones we expect.

Reply:
Looks good.